### ISSUE TITLE
discovery module feedback discussions

### ISSUE THREAD

**Issue by clux** [clux/kube-rs#523]
[discovery module](https://github.com/clux/kube-rs/blob/master/kube/src/client/discovery.rs) has been added in `0.54.0` under the `client` module, and it's the first release with pretty good support of everything (albeit its an early stage release)

here are a couple of things outstanding things I think we should do:

- [ ] probably deprecate old discovery utils like [`Client::list_*`](https://docs.rs/kube/0.54.0/kube/client/struct.Client.html#method.list_core_api_resources) to avoid [`APIResource`](https://arnavion.github.io/k8s-openapi/v0.11.x/k8s_openapi/apimachinery/pkg/apis/meta/v1/struct.APIResource.html) vs [`ApiResource`](https://docs.rs/kube-core/0.54.0/kube_core/api_resource/struct.ApiResource.html) confusion
- [x] add methods on `Discovery` to replace individual listers ([we can list any group](https://arnavion.github.io/k8s-openapi/v0.11.x/k8s_openapi/enum.GetNodeV1APIResourcesResponse.html?search=api_resources))
- ~~maybe move the discovery module to `api` ?~~ maybe not, now `api` is actually well defined and only for `Api`

But also potentially changing the interfaces a bit based on needs. Currently the module optimizes to simplify [dynamic_watcher](https://github.com/clux/kube-rs/blob/master/examples/dynamic_watcher.rs) and [dynamic_api](https://github.com/clux/kube-rs/blob/master/examples/dynamic_api.rs). If we are missing some __standard__ use case (like from the [client-go impl](https://github.com/kubernetes/client-go/blob/master/discovery/discovery_client.go)), and making them awkward, it would be very helpful to hear about it. Any particularly weird interfaces? It's got a bit more business logic than what is usual in kube because the Api is a bit awkward, but maybe we shouldn't be paving over some of them.

So in short, any feedback here is welcome.

**Comment by clux** [comment]
ok, have thought a lot more about this in #538. have yet to deprecate anything, but have changed all examples to use the new interfaces introduced therein. i think it's a lot cleaner, and will close this atm.
everything is in master, and will be released in next minor (i'll give it a day or two in master).
